### PR TITLE
Support to refresh the chatd/presenced URLs and reconnect

### PR DESCRIPTION
**Pull request by sergiohs84**
When API path is changed to staging, not only a full fetchnodes is required, but also the URL for presenced is different, so it needs to be re-fetched and reconnected.

### CODE REVIEW

**Comment by sergiohs84** [comment]
trigger compilation

File: src/chatd.cpp [hunk 1]
```diff
@@ -876,36 +868,69 @@ void Connection::doConnect()
...
+    }
 
-            setState(kStateDisconnected);
-            abortRetryController();
-            reconnect();
-        }
-        else if (mRetryCtrl && mRetryCtrl->state() == rh::State::kStateRetryWait)
+    if (refreshURL || !mUrl.isValid())
+    {
+        if (State::kStateFetchingUrl)
```
Comment (jgandres) [inline src/chatd.cpp]:
```suggestion
        if (mState == State::kStateFetchingUrl)
```

File: src/chatd.cpp [hunk 2]
```diff
@@ -876,36 +868,69 @@ void Connection::doConnect()
     }
 }
 
-void Connection::retryPendingConnection(bool disconnect)
+void Connection::retryPendingConnection(bool disconnect, bool refreshURL)
 {
-    if (mUrl.isValid())
+    if (mState == kStateNew)
```
Comment (jgandres) [inline src/chatd.cpp]:
```suggestion
    if (mState == State::kStateNew)
```

File: src/chatd.cpp [hunk 3]
```diff
@@ -876,36 +868,69 @@ void Connection::doConnect()
...
+
+            mUrl.parse(result->getLink());
+            retryPendingConnection(true);
+        });
+    }
+    else if (disconnect)
+    {
+        CHATDS_LOG_WARNING("retryPendingConnection: forced reconnection!");
+
+        setState(kStateDisconnected);
```
Comment (jgandres) [inline src/chatd.cpp]:
```suggestion
        setState(State::kStateDisconnected);
```

File: src/chatd.cpp (hunk 2 above)
Comment (sergiohs84) [reply src/chatd.cpp]: No need

**Pull request merged**